prowler/lib/outputs/finding.py

```diff
@@ -267,3 +274,183 @@ def generate_output(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
             raise error
+
+    @classmethod
+    def transform_api_finding(cls, finding) -> "Finding":
+        """
+        Transform a FindingModel instance into an API-friendly Finding object.
+
+        This class method extracts data from a FindingModel instance and maps its
+        properties to a new Finding object. The transformation populates various
+        fields including authentication details, timestamp, account information,
+        check metadata (such as provider, check ID, title, type, service, severity,
+        and remediation details), as well as resource-specific data. The resulting
+        Finding object is structured for use in API responses or further processing.
+
+        Args:
+            finding (API Finding): An API Finding instance containing data from the database.
+
+        Returns:
+            Finding: A new Finding instance populated with data from the provided model.
+        """
+        output_data = {}
+        output_data["auth_method"] = ""
```

Comment:
@jfagoagas review

Reply:
@AdriiiPRodri these are the contents for the `authe_method` field regarding each provider allowed in the API?
- GCP: ->`Principal: gcp_provider.identity.profile`, we need to get `gcp_provider.identity.profile`
- Azure -> `azure_provider.identity.identity_type: azure_provider.identity.identity_id`, we need to get `azure_provider.identity.identity_type` and  `azure_provider.identity.identity_id`
- K8s -> `kubeconfig`, since we don't support `in-cluster` which is the other `auth_method` supported.
- AWS -> `profile: None` since `profile` is not supported, in the future we'll need to adjust this but that's the format in the SDK

Reply:
`partition` is also missing and the content by provider is:

- GPC -> Nothing, so empty string
- Azure -> `{azure_provider.region_config.name}`
- K8s -> Nothing, so empty string
- AWS -> `{aws_provider.identity.partition}`